core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -302,8 +336,7 @@ public void transportTerminated() {
           log.warning("transportTerminated called without previous transportShutdown");
           activeTransport = null;
         }
-        // TODO(notcarl): replace this with something more meaningful
-        transportShutdown(Status.UNKNOWN);
+        clearActiveTransport();
```

Comment:
`transportShutdown()` was being called there before to simply guarantee that `transportShutdown()` was always called first. It didn't have anything to do with what `transportShutdown()` did other than gracefully handling a bug in the transport. When originally written, it wasn't clear transports would reliably call `transportShutdown()`, especially since there are multiple transport implementations.

Gracefully handling a bug in the transport isn't all that big of a deal, because all the transport are generally guaranteeing that `transportShutdown()` is called first, but we also don't have a 100% reliable mechanism to notice if a transport is doing the wrong thing.